### PR TITLE
Decouple cart templating methods from WC_Cart class

### PR DESCRIPTION
**Pull request by bor0**

### CODE REVIEW

File: templates/cart/cart.php [hunk 1]
```diff
@@ -56,7 +56,7 @@
 								// @codingStandardsIgnoreLine
```
Comment (mikejolley) [inline templates/cart/cart.php]: This template will need a version bump.

File: templates/cart/mini-cart.php [hunk 2]
```diff
@@ -43,7 +43,7 @@
 						<?php
```
Comment (mikejolley) [inline templates/cart/mini-cart.php]: This template will need a version bump.

File: includes/wc-template-functions.php [hunk 3]
```diff
@@ -2694,3 +2694,106 @@ function wc_page_noindex() {
...
+
+/**
+ * Gets and formats a list of cart item data + variations for display on the frontend.
+ *
+ * @since 3.3.0
+ * @param array $cart_item Cart item object.
+ * @param bool  $flat Should the data be returned flat or in a list.
+ * @return string
+ */
+function wc_get_cart_item_data( $cart_item, $flat = false ) {
```
Comment (mikejolley) [inline includes/wc-template-functions.php]: Should we rename this whilst at it? This function is getting a formatted list. `wc_get_formatted_cart_item_data`?

**Review by mikejolley — APPROVED** [review]
Some nitpicks.

@claudiulodro Can you make the call on whether we do this in 3.3?

**Comment by claudiulodro** [comment]
I'm OK with putting this in 3.3 assuming it's all ready by today or tomorrow.

This seems to have broken the "Undo".
To test:
1. Add item to cart.
2. Remove item from cart.
3. Click "Undo" link that appears when item is removed from cart.
4. Observe the item does not get added back to the cart.

**Comment by bor0** [comment]
Thanks @claudiulodro! Fixed Undo issue in b6801d376.

**Comment by claudiulodro** [comment]
I'll merge this once tests are finished or whoever sees this first can merge this if the tests are finished and look good.

**Comment by kloon** [comment]
Unit tests are fine, just PHPCS that failed, which is fine since it is related to the rest of the files and not the changes you made. Merging.

**Pull request merged**